core/src/apps/ethereum/sign_tx.py

```diff
@@ -130,9 +130,9 @@ async def confirm_tx_data(
         return
 
     # Handle ERC-20, currently only 'transfer' function
-    token, recipient, value = await handle_erc20_transfer(msg, defs, address_bytes)
+    token, recipient, value = await _handle_erc20_transfer(msg, defs, address_bytes)
 
-    if token is None and data_total_len > 0:
+    if data_total_len > 0:
```

Comment:
The change here and at line 146 seems to break the basic ERC-20 `transfer` contract. Compare the behavior of your branch with `main` with e.g. this command which initiates a send (`transfer` contract) of chainLINK token:
```shell
trezorctl ethereum sign-tx -n "m/44'/60'/0'/0/0" -d a9059cbb000000000000000000000000574bbb36871ba6b78e27f4b4dcfb76ea0091880b0000000000000000000000000000000000000000000000000000000000000123 -g 100 -G 100 -i 1 0x514910771AF9Ca656af840dff83E8264EcF986CA 0
```

Reply:
Good catch! I reverted that change.

Reply:
`transfer` seems to be working just fine now.